ip_cidr_set.go

```diff
@@ -0,0 +1,90 @@
+package main
```

Comment:
Could you possibly add some comments through this PR to explain what's going on? I'm finding it hard to follow with no comments

Also, do you think you could move this IP handling code to a `ip` package so that we don't keep it in main? I don't think it necessarily needs to be in main does it?

Reply:
I'll port this to the `ip` package after I rebase with master, which I'm dreading :sweat_smile: